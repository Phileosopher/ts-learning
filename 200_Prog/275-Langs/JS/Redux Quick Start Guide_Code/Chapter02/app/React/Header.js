import React from "react";
import Link from "react-router-dom/Link";

const Header = () => (
  <ul>
    <li>
      <Link to="/">Home</Link>
    </li>
    <li>
      <Link to="/about">Services</Link>
    </li>
    <li>
      <Link to="/topics">Contact Us</Link>
    </li>
    <li>
      <Link to="/topics">Login</Link>
    </li>
  </ul>
);

export default Header;
