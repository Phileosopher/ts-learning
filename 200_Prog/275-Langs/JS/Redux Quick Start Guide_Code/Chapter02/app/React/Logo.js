import React from "react";
import Link from "react-router-dom/Link";

const Logo = () => (
  <div>
    <Link to="/">FastDoctor</Link>
  </div>
);

export default Logo;
