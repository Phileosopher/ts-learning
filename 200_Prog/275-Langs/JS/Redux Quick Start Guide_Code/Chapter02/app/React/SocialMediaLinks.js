import React from "react";
import Link from "react-router-dom/Link";

const SocialMediaLinks = () => (
  <ul>
    <li>
      <Link to="https://www.facebook.com/jagapati">Facebook</Link>
    </li>
    <li>
      <Link to="https://github.com/sureshHARDIYA">GitHub</Link>
    </li>
    <li>
      <Link to="https://www.researchgate.net/profile/Suresh_Mukhiya2">
        ResearchGate
      </Link>
    </li>
    <li>
      <Link to="https://twitter.com/dr_code_skm">Twitter</Link>
    </li>
  </ul>
);

export default SocialMediaLinks;
