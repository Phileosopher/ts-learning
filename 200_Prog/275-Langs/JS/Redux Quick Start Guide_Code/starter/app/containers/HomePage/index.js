import React, { PureComponent } from "react";

/* eslint-disable react/prefer-stateless-function */
export default class HomePage extends PureComponent {
  render() {
    return <h1>This is the HomePage Redux-book container!</h1>;
  }
}
