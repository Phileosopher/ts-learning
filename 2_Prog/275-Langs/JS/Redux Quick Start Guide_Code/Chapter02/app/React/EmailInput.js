import React, { Component } from "react";

class EmailInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ""
    };
    this.handleEmailChange = this.handleEmailChange.bind(this);
  }

  handleEmailChange(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <div>
        <label>Enter Your Email</label>
        <input
          type="email"
          onChange={this.handleEmailChange}
          value={this.state.value}
          name="email"
        />
      </div>
    );
  }
}

export default EmailInput;
