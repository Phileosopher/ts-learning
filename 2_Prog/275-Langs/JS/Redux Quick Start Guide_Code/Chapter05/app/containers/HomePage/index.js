import React from 'react';
import { Route, Link } from 'react-router-dom';

const MenuLink = ({ to, ...rest }) => (
  <Route path={to}>
    {({ match }) => (
      <li className={match ? 'active' : ''}>
        <Link to={to} {...rest} />
      </li>
    )}
  </Route>
);

const Homepage = () => (
  <div>
    <h1>This is home page our our application.</h1>
    <ul>
      <MenuLink to="/about">About Us</MenuLink>
      <MenuLink to="/login">Login</MenuLink>
      <MenuLink to="/register">Register</MenuLink>
      <MenuLink to="/contact">Contact</MenuLink>
      <MenuLink to="/users/add">Add New User</MenuLink>
    </ul>
  </div>
);

export default Homepage;
