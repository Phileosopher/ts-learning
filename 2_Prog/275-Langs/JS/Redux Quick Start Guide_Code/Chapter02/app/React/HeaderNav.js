import React from "react";
import Link from "react-router-dom/Link";

import Header from "./Header";
import SocialMediaLinks from "./SocialMediaLinks";
import Logo from "./Logo";

const HeaderNav = () => (
  <div>
    <Logo />
    <Header />
    <SocialMediaLinks />
  </div>
);

export default HeaderNav;
