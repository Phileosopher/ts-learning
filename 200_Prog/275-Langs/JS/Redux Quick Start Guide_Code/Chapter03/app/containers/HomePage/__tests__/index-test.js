import React from "react";
import renderer from "react-test-renderer";
import HomePage from "../index";

jest.mock("react-router-dom", () => ({
  Link: "Link",
  Route: ({ children, path }) => children({ match: path === "/link-path" })
}));

it("should render correctly", () => {
  const component = renderer.create(<HomePage />);
  expect(component.toJSON()).toMatchSnapshot();
});
