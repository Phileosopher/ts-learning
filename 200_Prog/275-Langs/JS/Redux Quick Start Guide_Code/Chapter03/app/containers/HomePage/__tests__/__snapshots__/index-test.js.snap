// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <h1>
    This is home page our our application.
  </h1>
  <ul>
    <li
      className=""
    >
      <Link
        to="/about"
      >
        About Us
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        to="/login"
      >
        Login
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        to="/register"
      >
        Register
      </Link>
    </li>
    <li
      className=""
    >
      <Link
        to="/contact"
      >
        Contact
      </Link>
    </li>
  </ul>
</div>
`;
